File: MyAPI/MyAPI/Data/DiscountCode.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace MyAPI.Data
{
    public class DiscountCode
    {
        public int Id { get; set; }
        [ForeignKey(nameof(Order))]
        public int? OrderId { get; set; }
        public virtual Order Order { get; set; }
        public string Code { get; set; }
        public string DiscountPercent { get; set; }
        public string DiscountAmount { get; set; }
        public string StartDate { get; set; }
        public string EndDate { get; set; }
        public int Status { get; set; }
    }
}

File: MyFE/src/app/class/discount-code.ts
import { Order } from "./order"

export class DiscountCode {
    id!:number
    order!:Order
    orderId!:number
    discountAmount!:string
    discountPercent!: string
    status!:number
    code!:string
    startDate!:string 
    endDate!:string 
}
